### ISSUE TITLE
Add example script for digital ocean

### ISSUE THREAD

**Issue by kris-nova** [kris-nova/kubicorn#146]
We offer example scripts for users who want to use kubicorn as a library. We have one for AWS here
https://github.com/kris-nova/kubicorn/blob/master/examples/aws.go

Can we please make one for digital ocean?

**Comment by MyDigitalLife** [comment]
I've added a digital ocean example and created a pull request #148. I have however not been able to test this because i could not find any documentation no how to use this. If you could point me in the right direction i would test it fully.

**Comment by kris-nova** [comment]
You should just be able to `go run` the file

**Comment by MyDigitalLife** [comment]
Looks like there is still a problem here. When running `go run basiccluster.go` i get the following output:

````
2017-08-05T22:48:15+02:00 [✔]  Init Cluster
2017-08-05T22:48:15+02:00 [▶]  Running preprocessors
2017-08-05T22:48:15+02:00 [▶]  ssh.Expected
2017-08-05T22:48:15+02:00 [▶]  ssh.Render
2017-08-05T22:48:15+02:00 [▶]  droplet.Expected
2017-08-05T22:48:15+02:00 [▶]  droplet.Render
2017-08-05T22:48:15+02:00 [▶]  droplet.Expected
2017-08-05T22:48:15+02:00 [▶]  droplet.Render
2017-08-05T22:48:15+02:00 [▶]  ssh.Actual
2017-08-05T22:48:15+02:00 [▶]  ssh.Render
2017-08-05T22:48:15+02:00 [▶]  droplet.Actual
2017-08-05T22:48:15+02:00 [▶]  droplet.Render
2017-08-05T22:48:15+02:00 [▶]  droplet.Actual
2017-08-05T22:48:16+02:00 [▶]  droplet.Render
2017-08-05T22:48:16+02:00 [▶]  ssh.Expected
2017-08-05T22:48:16+02:00 [▶]  Using cached ssh [expected]
2017-08-05T22:48:16+02:00 [▶]  ssh.Actual
2017-08-05T22:48:16+02:00 [▶]  Using cached ssh [actual]
2017-08-05T22:48:16+02:00 [▶]  droplet.Apply
2017-08-05T22:48:16+02:00 [▶]  Actual   : 3aa0d9d5c8e4d36d844617b475d278fc
2017-08-05T22:48:16+02:00 [▶]  Expected : 76476dc182047a11b96afb03f50093aa
2017-08-05T22:48:16+02:00 [✔]  Created SSH Key [11664620]
2017-08-05T22:48:16+02:00 [▶]  ssh.Render
2017-08-05T22:48:16+02:00 [▶]  droplet.Expected
2017-08-05T22:48:16+02:00 [▶]  Using droplet subnet [expected]
2017-08-05T22:48:16+02:00 [▶]  droplet.Actual
2017-08-05T22:48:16+02:00 [▶]  Using cached droplet [actual]
2017-08-05T22:48:16+02:00 [▶]  droplet.Apply
2017-08-05T22:48:16+02:00 [▶]  Actual   : 434e48409fae71cae25365cb9b27b321
2017-08-05T22:48:16+02:00 [▶]  Expected : 0bdaa31228b9fb5110fbe07b56c83f2e
2017-08-05T22:48:17+02:00 [✖]  Error during apply! Attempting cleaning: POST https://api.digitalocean.com/v2/droplets: 422 There was an error while trying to create your tags. Please try again.
2017-08-05T22:48:17+02:00 [!]  --------------------------------------
2017-08-05T22:48:17+02:00 [!]  Attempting to delete created resources!
2017-08-05T22:48:17+02:00 [!]  --------------------------------------
2017-08-05T22:48:17+02:00 [▶]  ssh.Delete
2017-08-05T22:48:17+02:00 [✔]  Deleted SSH Key [11664620]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb58f74]

goroutine 1 [running]:
main.main()
        /home/peter/goprojects/src/github.com/mydigitallife/kubicorn/examples/digitalocean/basiccluster.go:51 +0x174
exit status 2
````

**Comment by xmudrii** [comment]
@MyDigitalLife do you already have that tag created? It could be similar issue to the SSH keys. We should make sure that if resources such as keys and tags exists, they should be used rather than error.

**Comment by MyDigitalLife** [comment]
I had some droplets that have had the tags but they where already deleted at that point. I don't really see any existing list of tags in the web UI. In the Digital ocean documentation it does look like it has a list of tag saved and it is probably throwing the error because they already exist like you said. 

I'l try and fix this tomorrow. We should first check what tags exist and use those before trying to make new ones.

**Comment by xmudrii** [comment]
That's true, even when you delete a Droplet, tags are not deleted.
You have to use [this](https://developers.digitalocean.com/documentation/v2/#list-all-tags) API endpoint to list all tags and then [this](https://developers.digitalocean.com/documentation/v2/#delete-a-tag) one to delete.
Or you can use CLI - [`doctl`](do.co/doctl) to list and delete them (`doctl compute tag list`, `doctl compute tag delete`).

If you need any help, comment here and I'll try to take a look tomorrow.

**Comment by kris-nova** [comment]
Where are we on this? Is this still blocking?

**Comment by MyDigitalLife** [comment]
I just tested this and found a few problems. I don't the error i was getting before. But i did find these 2 problems:

- Meshbird does not work anymore. The changes to make openvpn broke the code for meshbird
- Deleting of droplets does not work. Its just giving `Droplet count mis-match, trying query again`

**Comment by xmudrii** [comment]
Closed by #148
